plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -0,0 +1,290 @@
+      @Override
+      public void modifyText(ModifyEvent event) {
+        projectSelector.setFilter(filterField.getText());
+      }
+    });
+
+    // Service key row
+    new Label(composite, SWT.LEAD).setText(Messages.getString("label.service.key")); //$NON-NLS-1$
+    serviceKeyInput = new Text(composite, SWT.BORDER);
+    serviceKeyInput.addModifyListener(new ModifyListener() {
+      @Override
+      public void modifyText(ModifyEvent event) {
+        updateLaunchConfigurationDialog();
+      }
+    });
+    Button browse = new Button(composite, SWT.NONE);
+    browse.setText(Messages.getString("button.browse")); //$NON-NLS-1$
+
+    GridDataFactory.fillDefaults().align(SWT.BEGINNING, SWT.TOP).applyTo(projectLabel);
+    GridDataFactory.fillDefaults().span(2, 1).applyTo(accountSelector);
+    GridLayoutFactory.swtDefaults().numColumns(3).generateLayout(composite);
+
+    GridDataFactory.fillDefaults().span(2, 1).applyTo(projectSelectorComposite);
+    GridDataFactory.fillDefaults().applyTo(filterField);
+    GridDataFactory.fillDefaults().grab(true, false).hint(SWT.DEFAULT, 200)
+        .applyTo(projectSelector);
+    GridLayoutFactory.fillDefaults().spacing(0, 0).generateLayout(projectSelectorComposite);
+
+    setControl(composite);
+
+    gcpIcon = SharedImages.GCP_IMAGE_DESCRIPTOR.createImage();
+    composite.addDisposeListener(new ImageDisposer(gcpIcon));
+  }
+
+  private void updateProjectSelector() {
+    Credential credential = accountSelector.getSelectedCredential();
+    if (credential == null) {
+      projectSelector.setProjects(new ArrayList<GcpProject>());
+    } else {
+      try {
+        List<GcpProject> gcpProjects = projectRepository.getProjects(credential);
+        projectSelector.setProjects(gcpProjects);
+      } catch (ProjectRepositoryException e) {
+        logger.log(Level.WARNING,
+            "Could not retrieve GCP project information from server.", e); //$NON-NLS-1$
+      }
+    }
+  }
+
+  @Override
+  public void setDefaults(ILaunchConfigurationWorkingCopy configuration) {
+    // No particular default values to set in a newly created configuration.
+  }
+
+  @Override
+  public void activated(ILaunchConfigurationWorkingCopy workingCopy) {
+    activated = true;
+    super.activated(workingCopy);
+  }
+
+  @Override
+  public void deactivated(ILaunchConfigurationWorkingCopy workingCopy) {
+    super.deactivated(workingCopy);
+    activated = false;
+  }
+
+  @Override
+  public void initializeFrom(ILaunchConfiguration configuration) {
+    accountEmailModel = getAttribute(configuration, ATTRIBUTE_ACCOUNT_EMAIL, ""); //$NON-NLS-1$
+
+    Map<String, String> environmentMap = getEnvironmentMap(configuration);
+    gcpProjectIdModel = Strings.nullToEmpty(environmentMap.get(PROJECT_ID_ENVIRONMENT_VARIABLE));
+    String serviceKey = Strings.nullToEmpty(environmentMap.get(SERVICE_KEY_ENVIRONMENT_VARIABLE));
+
+    initializingUiValues = true;
+    accountSelector.selectAccount(accountEmailModel);
```

Comment:
I think it's worth mentioning here that the account-selection-event listener updates the project list and waits until complete.  Which isn't great for UI liveness, but …